h5py/_objects.templ.pyx

```diff
@@ -222,21 +222,25 @@ cdef class ObjectID:
 
 
     def __dealloc__(self):
+        self._dealloc()
+
+    # During interpreter shutdown, module attributes are set to None
+    # before __dealloc__ and __del__ methods are executed.
+    def _dealloc(self, _phil=_phil, warn=warnings.warn, registry=registry):
```

Comment:
I continue to have doubts that we need this for `registry` and `_phil` in the non-freethreaded case (as they are `cdef`'d and to my understanding not managed/accessible to Python so I don't think Python's shut-down order can mess with us).

However if Cython is happy to compile this I do not see any harm in being extra defensive and forcing them to be locals.